main/main.cpp

```diff
@@ -1306,14 +1309,37 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 
 	// possibly be worth changing the default from vulkan to something lower spec,
 	// for the project manager, depending on how smooth the fallback is.
-	GLOBAL_DEF_RST("rendering/driver/driver_name", "vulkan");
 
 	// this list is hard coded, which makes it more difficult to add new backends.
 	// can potentially be changed to more of a plugin system at a later date.
+#ifdef VULKAN_ENABLED
+	defaultRenderer = "vulkan";
+
+#ifdef GLES3_ENABLED
+	rendererHints = "vulkan,opengl3";
+#else
+	rendererHints = "vulkan";
+#endif
+
+#else
+
+#ifdef GLES3_ENABLED
+	defaultRenderer = "opengl3";
+	rendererHints = "opengl3";
+#else
+	// no known rendering driver enabled; should this be a warning?
+	defaultRenderer = "";
+	rendererHints = "";
+#endif
+
+#endif
+
+	GLOBAL_DEF_RST("rendering/driver/driver_name", defaultRenderer);
```

Comment:
Maybe simpler?
```suggestion
String renderer_hint;

// Start with Vulkan, which will be the default if enabled.
#ifdef VULKAN_ENABLED
	renderer_hint = "vulkan";
#endif

// And OpenGL3 next, or first if Vulkan is disabled.
#ifdef GLES3_ENABLED
	if (!renderer_hint.is_empty()) {
		renderer_hint += ",";
	}
	renderer_hint += "opengl3";
#endif

if (renderer_hint.is_empty()) {
	ERR_PRINT("No rendering driver available.")
}

String default_renderer = renderer_hint.slice(",", 0);
GLOBAL_DEF_RST("rendering/driver/driver_name", default_renderer);
```

Reply:
is that going to compile?   are you allowed to goto over that stack declaration?

Reply:
What makes this no longer fun is that what I provided is much better than what was there (failure to consider the conditional compilation flags at all.)   Because I tried to fix that, I am now stuck with all this nitpicking.   I guess if I had made separate PRs then I could have just withdrawn the ones that aren't worth it, but I was really trying to get out the important material changes that are hurting opengl3 development that is happening right now.   So no, I don't care even a tiny bit about how nice this temporary hack of hard coded list is.  It's just not worth my time to work on that.  And I understand now that this means this code base and I aren't going to get along. :)    And that's fine too.

Reply:
@akien-mga I will finish this PR (today) so that we can put this all behind us :).  Please be specific on what you need to accept it.

Reply:
I put in your code, even though I don't believe String+= and get_slice are a simpler solution than conditionally compiling a single string constant to different values.  I think the best would actually be to but the #ifdef tree at the top of the function and just set a single const char* so it is static data in a data segment.   That would be far away from where it is used, which I didn't like originally.

Reply:
Correction: it is also static data in the code I had that you didn't like, it just gets converted to a String object, which makes sense I think in the code base.